core/src/main/java/tech/pegasys/web3signer/core/util/OpenApiSpecsExtractor.java

```diff
@@ -0,0 +1,206 @@
+    for (Map.Entry<String, Object> entry : yamlMap.entrySet()) {
+      final String key = entry.getKey();
+      final Object value = entry.getValue();
+
+      if ("$ref".equals(key) && value instanceof String) {
+        modifyRefValue(yamlFile, entry);
+      } else if ("discriminator".equals(key) && value instanceof Map) {
+        if (((Map<String, Object>) value).containsKey("mapping")) {
+          Map<String, Object> discriminatorMappings =
+              (Map<String, Object>) ((Map<String, Object>) value).get("mapping");
+          discriminatorMappings
+              .entrySet()
+              .forEach(discriminatorMapEntry -> modifyRefValue(yamlFile, discriminatorMapEntry));
+        }
+      } else if (value instanceof List) {
+        List listItems = (List) value;
+        for (Object listItem : listItems) {
+          if (listItem instanceof Map) {
+            fixRelativePathInOpenApiMap(yamlFile, (Map<String, Object>) listItem);
+          } // else we are not interested
+        }
+      } else if (value instanceof Map) {
+        fixRelativePathInOpenApiMap(yamlFile, (Map<String, Object>) value);
+      }
+    }
+  }
+
+  private void modifyRefValue(final Path yamlFile, final Map.Entry<String, Object> entry) {
+    final String fileName = StringUtils.substringBefore(entry.getValue().toString(), "#");
+    final String jsonPointer = StringUtils.substringAfter(entry.getValue().toString(), "#");
+    if (StringUtils.isNotBlank(fileName)) {
+      // convert filename path to absolute path
+      final Path parent = yamlFile.getParent();
+      final Path nonNormalizedRefPath = parent.resolve(Path.of(fileName));
+      // remove any . or .. from path
+      final Path normalizedPath = nonNormalizedRefPath.toAbsolutePath().normalize();
+
+      final String updatedValue =
+          StringUtils.isBlank(jsonPointer)
+              ? normalizedPath.toString()
+              : normalizedPath + "#" + jsonPointer;
+      entry.setValue(updatedValue);
+    }
+  }
+
+  private List<Path> extractToDestination() throws IOException {
+    final List<Path> extractedResources = new ArrayList<>();
+
+    try (final Stream<URL> webrootYamlFiles =
+        Resources.find("/" + OPENAPI_RESOURCES_ROOT + "**.*")) {
+      webrootYamlFiles.forEach(
+          openapiResource -> {
+            final String jarPath = openapiResource.getPath();
+            // the path is in file://.jar!<path> format. We need the non-jar path.
+            final String filePath =
+                StringUtils.substringAfter(jarPath, "!/" + OPENAPI_RESOURCES_ROOT + "/");
+            final Path extractedResource = destinationDirectory.resolve(filePath);
+            copyContents(openapiResource, extractedResource);
+            extractedResources.add(extractedResource);
+          });
+    }
+    return Collections.unmodifiableList(extractedResources);
+  }
+
+  private static void copyContents(final URL openapiResource, final Path extractedResource) {
```

Comment:
Could we avoid copying files? I think OpenApiV3Parser takes a string and we are also returning a string to Vertx.

Reply:
This doesn't need to be static

Reply:
we need to extract resource files from jar to local dist.